domain_config_ui/src/DomainConfigUINegotiator.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Drupal\domain_config_ui;
+
+use Drupal\domain\DomainNegotiator;
+use Drupal\domain_config\DomainConfigOverrider;
+
+/**
+ * {@inheritdoc}
+ */
+class DomainConfigUINegotiator extends DomainNegotiator {
+  /**
+   * @var DomainConfigOverrider
+   */
+  protected $domainConfigOverrider;
+
+  /**
+   * Determine the active domain.
+   */
+  protected function negotiateActiveDomain() {
+    // Set http host to be that of the selected domain to configure.
+    if ($selected_domain = $this->getSelectedDomain()) {
+      $httpHost = $selected_domain->getHostname();
+    }
+    else {
+      $httpHost = $this->negotiateActiveHostname();
+    }
+    $this->setRequestDomain($httpHost);
```

Comment:
I don't think spoofing the request domain to load is the correct solution - will try to move this to a better place